### PR TITLE
Update src/links js comments

### PR DESCRIPTION
**Pull request by janbrasna**
While exploring the details of autolinking, noticed some comments and console warnings in scripts have some more typos and such not fixed in #30894

Fixing in scripts that are not workflow-protected.

### Why:

Followup of: #30895

### What's being changed (if available, include any code snippets, screenshots, or gifs):

- Mostly comments.
- One output warning grammar. _(The string not found elsewhere so is not probably used anywhere in tests…)_

### Check off the following:

- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

File: src/links/lib/update-internal-links.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ import getRedirect, { splitPathByLanguage } from '#src/redirects/lib/get-redirec
 import nonEnterpriseDefaultVersion from '#src/versions/lib/non-enterprise-default-version.js'
 import { deprecated } from '#src/versions/lib/enterprise-server-releases.js'
 
-// That magical string that can be turned into th actual title when
+// That magical string that can be turned into the actual title when
 // we, at runtime, render out the links
 const AUTOTITLE = 'AUTOTITLE'
```
Comment (rodneymorgan225) [inline src/links/lib/update-internal-links.js:25]: - [ ] - []()

File: src/links/lib/update-internal-links.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ import getRedirect, { splitPathByLanguage } from '#src/redirects/lib/get-redirec
 import nonEnterpriseDefaultVersion from '#src/versions/lib/non-enterprise-default-version.js'
 import { deprecated } from '#src/versions/lib/enterprise-server-releases.js'
 
-// That magical string that can be turned into th actual title when
+// That magical string that can be turned into the actual title when
```
Comment (rodneymorgan225) [inline src/links/lib/update-internal-links.js:23]:
```suggestion
- // That magical string that can be turned into the actual title when@__
```

File: src/links/lib/update-internal-links.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ import getRedirect, { splitPathByLanguage } from '#src/redirects/lib/get-redirec
 import nonEnterpriseDefaultVersion from '#src/versions/lib/non-enterprise-default-version.js'
```
Comment (rodneymorgan225) [inline src/links/lib/update-internal-links.js:20]: - [ ]

**Review by rodneymorgan225 — COMMENTED** [review]
Done

**Comment by janbrasna** [comment]
There's one more but that location is probably guarded by the spelunker script;)

```diff
diff --git a/src/links/scripts/update-internal-links.js b/src/links/scripts/update-internal-links.js
index 38a1081ff3a91..d1f9e2d111ea9 100755
--- a/src/links/scripts/update-internal-links.js
+++ b/src/links/scripts/update-internal-links.js
@@ -185,7 +185,7 @@ async function main(files, opts) {
 
       if (countWarnings > 0) {
         console.log(chalk.yellow('\nNote! Warnings can currently not be automatically fixed.'))
-        console.log('Manually edit heeded warnings and run the script again to update.')
+        console.log('Manually edit needed warnings and run the script again to update.')
       }
 
       if (countChangedFiles > 0) {
```

(Not sure if "heeded" was an intention, but might be…)

**Review by nguyenalex836 — APPROVED** [review]

**Comment by nguyenalex836** [comment]
@janbrasna Thanks so much for opening a PR with this fix! I'll update the branch and get this merged once tests are passing 🍏

> There's one more but that location is probably guarded by the spelunker script;)

Appreciate the callout ✨ I'll check with the team, and we'll open a PR internally to fix if needed 💛

**Pull request merged**